anchore_engine/auth/aws_ecr.py

```diff
@@ -54,7 +54,11 @@ def refresh_ecr_credentials(registry, access_key_id, secret_access_key):
         elif access_key_id == '_iam_role':
             try:
                 sts = boto3.client('sts')
-                session = sts.assume_role(RoleArn=secret_access_key, RoleSessionName=str(int(time.time())))
+                role_arn, external_id = secret_access_key.split(";") # pass = "role_arn;external_id"
```

Comment:
This will throw a ValueError if the secret_access_key value does not contain a ';' due to the tuple assignment. If you assign the split output to a single (list) value then test the length of the list for existence of the 2nd element then it should be safer.

e.g. 
```
components = secret_access_key.split(';')
role_arn = components[0]
if len(components) > 1:
  external_id = components[1]
```

Reply:
Thanks for the input. Will make these changes asap.